iis-wf/iis-wf-import/src/main/java/eu/dnetlib/iis/wf/importer/infospace/ImportInformationSpaceJob.java

```diff
@@ -0,0 +1,404 @@
+            JavaRDD<DocumentToProject> docProjRelation = filterAndParseRelationToAvro(sourceRelation, resultApprover, docProjectConverter, 
+                    REL_TYPE_RESULT_PROJECT, SUBREL_TYPE_OUTCOME, REL_NAME_IS_PRODUCED_BY);
+            JavaRDD<ProjectToOrganization> projOrgRelation = filterAndParseRelationToAvro(sourceRelation, resultApprover, projectOrganizationConverter, 
+                    REL_TYPE_PROJECT_ORGANIZATION, SUBREL_TYPE_PARTICIPATION, REL_NAME_HAS_PARTICIPANT); 
+            JavaRDD<IdentifierMapping> dedupRelation = filterAndParseRelationToAvro(sourceRelation, resultApprover, deduplicationMappingConverter, 
+                    REL_TYPE_RESULT_RESULT, SUBREL_TYPE_DEDUP, REL_NAME_MERGES); 
+            
+            storeInOutput(sc, docMeta, dataset, project, organization, docProjRelation, projOrgRelation, dedupRelation, params);
+            
+        } finally {
+            if (Objects.nonNull(session) && !isSparkSessionManagedOutside) {
+                session.stop();
+            }
+        }
+    }
+    
+    /**
+     * Parses given set of RDDs conveying various {@link Result} entities into a single RDD with {@link DocumentMetadata} records.
+     */
+    private static JavaRDD<eu.dnetlib.iis.importer.schemas.DocumentMetadata> parseToDocMetaAvro(
+            JavaRDD<eu.dnetlib.dhp.schema.oaf.Dataset> filteredDataset,
+            JavaRDD<eu.dnetlib.dhp.schema.oaf.Publication> sourcePublication,
+            JavaRDD<eu.dnetlib.dhp.schema.oaf.Software> sourceSoftware,
+            JavaRDD<eu.dnetlib.dhp.schema.oaf.OtherResearchProduct> sourceOtherResearchProduct,
+            ResultApprover resultApprover, OafEntityToAvroConverter<Result, DocumentMetadata> documentConverter) {
+        JavaRDD<eu.dnetlib.iis.importer.schemas.DocumentMetadata> publicationDocMeta = filterAndParseResultToAvro(
+                sourcePublication, resultApprover, documentConverter);
+
+        JavaRDD<eu.dnetlib.iis.importer.schemas.DocumentMetadata> softwareDocMeta = filterAndParseResultToAvro(
+                sourceSoftware, resultApprover, documentConverter);
+
+        JavaRDD<eu.dnetlib.iis.importer.schemas.DocumentMetadata> orpDocMeta = filterAndParseResultToAvro(
+                sourceOtherResearchProduct, resultApprover, documentConverter);
+
+        JavaRDD<eu.dnetlib.iis.importer.schemas.DocumentMetadata> datasetDocMeta = parseResultToAvro(
+                filteredDataset, documentConverter);
+
+        return publicationDocMeta.union(softwareDocMeta).union(orpDocMeta).union(datasetDocMeta);
+    }
+    
+    /**
+     * Parses given RDD with {@link OafEntity} entities into {@SpecificRecord} avro RDD.
+     */
+    private static <S extends OafEntity, T extends SpecificRecord>JavaRDD<T> filterAndParseToAvro(JavaRDD<S> source, 
+            ResultApprover resultApprover, OafEntityToAvroConverter<S, T> entityConverter) {
+        return parseToAvro(source.filter(x -> resultApprover.approve(x)), entityConverter);
```

Comment:
Yes, let's follow the hint. 

I might be accustomed to the former notion too much, mainly because it explicitly shows what is going to be performed and it seems to be easier to read (at least for me). But maybe I am the only one bound to some old-fashioned conventions.

Reply:
I also use the old style when writing the functions and let IDE figure out how to make it more compact.